simplesam.py

```diff
@@ -237,7 +226,7 @@ def __init__(self, f, header=None):
         try:
             _, ext = os.path.splitext(f.name)
             if ext == '.bam':
-                raise NotImplementedError('Bam writing support is not implemented.\n')
+                raise NotImplementedError('Bam writing support is not implemented.\n') # why not just pipe to samtools?
```

Comment:
That was originally the thought, but if I remember correctly there were some issues with piping to some samtools versions at the time. It's worth implementing now and making sure it works with a recent samtools build.